src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -0,0 +1,130 @@
+package seedu.address.model.session;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+/**
+ * Class that handles RecurringSession that extend Session.
+ */
+public class RecurringSession extends Session {
+    public static final String MESSAGE_CONSTRAINTS = "End date/time does not match starting date and interval.";
+    private Interval interval;
+    private SessionDate endDateTime;
+
+    /**
+     * Constructs a {@code RecurringSession}.
+     * <p>
+     * Requires that all fields entered to be non null
+     *
+     * @param sessionDate
+     * @param duration
+     * @param subject
+     * @param fee
+     * @param interval
+     * @param endDateTime must match sessionDate and interval.
+     */
+    public RecurringSession(SessionDate sessionDate, Duration duration, Subject subject, Fee fee,
+                            Interval interval, SessionDate endDateTime) {
+        super(sessionDate, duration, subject, fee);
+        requireAllNonNull(interval, endDateTime);
+        this.interval = interval;
+        checkArgument(isValidEnd(sessionDate, endDateTime, interval), MESSAGE_CONSTRAINTS);
+        this.endDateTime = endDateTime;
```

Comment:
Hi, just checking, for this implementation, we run super(session) to create a standalone session out, but then it doesn't seem to keep a reference to the created session anymore. Can I check if that is the intended effect?

Reply:
No it's not intended because I didn't (and kind of currently dont) know what problems this may cause.
What problems might this cause/ what recommended way might there be?
Was thinking since its extending Session, the parent's method to access Sesssion contents are inherited

Reply:
Hmmmm already, actually didn't use super often so not sure how it would work out. We could keep this in mind and when the rest of us work on features that uses recurring session, see if we are able to perform all the actions we want, even w/o keeping a reference to it ba.